hil-test/tests/gpio_custom_handler.rs

```diff
@@ -71,6 +80,14 @@ mod tests {
         let timg0 = TimerGroup::new(peripherals.TIMG0);
         esp_hal_embassy::init(timg0.timer0);
 
+        // We need to enable the GPIO interrupt, otherwise the async Future's
+        // setup or Drop implementation hangs.
+        esp_hal::interrupt::enable(
```

Comment:
Is this specific to the test case, or is this something we should document in `Io` too?

Reply:
Hmm, I suppose it's quite an advanced use-case, I'm not entirely sure where to put it either tbh

Reply:
PTAL at the latest commit.